### PR TITLE
added common mistakes for `generics` tasks

### PR DESCRIPTION
**Pull request by KseniiaMakarova**

### CODE REVIEW

File: java-core/generics/storage.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### Common mistakes
+
+#### All non-constant fields should be initialized in the constructor
+Example:
+```java
+public class Shelf {
+    private Object[] items;
+
+    public Shelf {
```
Comment (boroda4436) [inline java-core/generics/storage.md:9]:
```suggestion
    public Shelf() {
```

File: java-core/generics/storage.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        items = new Object[]{"book", "photo", "phone"};
+    }
+}
+```
+#### Any magic numbers should be constants
+Your code should be easy to read. Let's move all hardcoded values to constant fields and give them informative names.
```
Comment (boroda4436) [inline java-core/generics/storage.md:15]: Can we add good and bad examples?

**Review by boroda4436 — APPROVED** [review]

File: java-core/generics/storage.md (hunk 2 above)
Comment (KseniiaMakarova) [reply java-core/generics/storage.md:15]: sure

File: java-core/generics/storage.md (hunk 1 above)
Comment (KseniiaMakarova) [reply java-core/generics/storage.md]: oops

**Pull request merged**